lib/instrumentation/index.ts

```diff
@@ -0,0 +1,91 @@
+import { ClientRequest } from 'http';
+import type { Tracer } from '@opentelemetry/api';
+import * as api from '@opentelemetry/api';
+import { SpanStatusCode } from '@opentelemetry/api';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { BunyanInstrumentation } from '@opentelemetry/instrumentation-bunyan';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { Resource } from '@opentelemetry/resources';
+import {
+  BatchSpanProcessor,
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { pkg } from '../expose.cjs';
+import {
+  isTraceDebuggingEnabled,
+  isTraceSendingEnabled,
+  isTracingEnabled,
+} from './utils';
+
+let traceProvider: NodeTracerProvider;
+
+init();
```

Comment:
Is it intentional that this method is called when this file is imported? I'd prefer that it need to be manually initialized, instead of every time `instrumentation` is imported.

Reply:
Yes, this is intentional as the instrumentation code has to run before the bunyan is loaded. 
https://github.com/secustor/renovate/blob/c45cb7fd6888b5e9f89e980e2d283a8ae52d2c9f/lib/renovate.ts#L3-L4

Not sure how to prevent this without removing the logging from `renovate.ts`

Reply:
so then we should call this from `renovate.ts` before we start logging?

we should also add an init method for logging, so we can defer / lazy initialize logging after instrumentation?

Reply:
The instrumentation has to be run before even the instrumented packages are loaded. 
https://opentelemetry.io/blog/2022/troubleshooting-nodejs/#enable-before-require    

So from my understanding simply creating an init function for logging will be not enough.

What do you think about this: 
```suggestion
if (!traceProvider) {
  init();
}
```
that way it gets only initialized during the first import.

Reply:
Yeah, I'd be okay with that.

Could you also add a global mock for tests so the import of this module is a noop? Then do a `jest.requireActual` for specific tests?

Reply:
good idea, checkout logger / platform tests for reset global mock sample

Reply:
I have added a global mock. 

> Is it intentional that this method is called when this file is imported? I'd prefer that it need to be manually initialized, instead of every time instrumentation is imported.

I have tried to confirm that it is executed everytime the module is imported, but it has been only executed once.  Seems like this expected behavior:
https://stackoverflow.com/questions/65638667/does-importingrequire-module-in-js-also-executes-whole-code-as-well